src/irmin/tree_intf.ml

```diff
@@ -82,6 +82,12 @@ module type S = sig
   (** [is_empty t] is true iff [t] is {!empty} (i.e. a tree node with no
       children). Trees with {!kind} = [`Contents] are never considered empty. *)
 
+  val is_val : t -> path -> bool
+  (** [is_val t k] is [true] iff the path [k] has already been forced in [t]. In
+      that case, that means that all the nodes traversed by [k] are loaded in
+      memory. If the leaf node is a contents [c], then [Contents.is_val c]
+      should also be [true]. *)
```

Comment:
maybe the doc should also mention that it returns true also when there is no node at the path given